manager/src/test/java/com/redhat/service/bridge/manager/workers/resources/AbstractWorkerTest.java

```diff
@@ -5,52 +5,107 @@
 import com.redhat.service.bridge.test.resource.PostgresResource;
 
 import io.quarkus.test.common.QuarkusTestResource;
 import io.quarkus.test.junit.QuarkusTest;
 import io.quarkus.test.junit.mockito.InjectMock;
 
+import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
-import static org.mockito.Mockito.verify;
 
 @QuarkusTest
 @ExtendWith(MockitoExtension.class)
 @QuarkusTestResource(PostgresResource.class)
 public class AbstractWorkerTest {
 
-    private static final String RESOURCE_ID = "123";
-
-    @InjectMock
-    WorkManager workManager;
-
     @InjectMock
     @SuppressWarnings("unused")
     //Needed to set up RHOAS for tests
     RhoasService rhoasService;
 
+    @Inject
+    WorkManager workManager;
+
     @Inject
     BridgeWorker worker;
 
+    @Inject
+    BridgeDAO bridgeDAO;
+
+    @Inject
+    DatabaseManagerUtils databaseManagerUtils;
+
+    @ConfigProperty(name = "event-bridge.resources.worker.max-retries")
+    int maxRetries;
+
+    @ConfigProperty(name = "event-bridge.resources.workers.timeout-seconds")
+    int timeoutSeconds;
+
+    @BeforeEach
+    public void setup() {
+        databaseManagerUtils.cleanUp();
+    }
+
     @Test
     @Transactional
     void workIsCompletedWhenResourceIsNotFound() {
-        Work work = new Work();
-        work.setType(Bridge.class.getName());
-        work.setSubmittedAt(ZonedDateTime.now());
-        work.setManagedResourceId(RESOURCE_ID);
+        Bridge bridge = Fixtures.createBridge();
+        // It is intentional that we do not persist the Bridge here
+        Work work = workManager.schedule(bridge);
+        assertThat(workManager.exists(work)).isTrue();
 
         assertThatThrownBy(() -> worker.handleWork(work)).isInstanceOf(IllegalStateException.class);
+    }
+
+    @Test
+    @Transactional
+    void workIsCompletedWhenMaxRetriedExceeded() {
+        Bridge bridge = Fixtures.createBridge();
+        // Persist Bridge so that it can be found by the Worker
+        bridgeDAO.persist(bridge);
+
+        Work work = workManager.schedule(bridge);
+        assertThat(workManager.exists(work)).isTrue();
+        work.setAttempts(maxRetries + 1);
+
+        worker.handleWork(work);
+
+        assertThat(bridge.getStatus()).isEqualTo(ManagedResourceStatus.FAILED);
+
+        assertThat(workManager.exists(work)).isFalse();
+    }
+
+    @Test
+    @Transactional
+    void workIsCompletedWhenTimedOut() {
+        Bridge bridge = Fixtures.createBridge();
+        // Persist Bridge so that it can be found by the Worker
+        bridgeDAO.persist(bridge);
+
+        Work work = workManager.schedule(bridge);
+        assertThat(workManager.exists(work)).isTrue();
+        work.setSubmittedAt(ZonedDateTime.now().minusSeconds(timeoutSeconds * 2L));
+
+        worker.handleWork(work);
+
+        assertThat(bridge.getStatus()).isEqualTo(ManagedResourceStatus.FAILED);
 
-        verify(workManager).complete(work);
```

Comment:
Because it is no longer a mock and I now check if the work was removed from the queue instead.

The `complete(..)` method deletes the work from the queue (and yes, I could have spied instead but @robpblake is advocating less use of mocks so I relied on the actual implementation instead).

Reply:
Thanks